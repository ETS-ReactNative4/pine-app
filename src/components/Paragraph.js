import React, { Component } from 'react';
import { StyleSheet } from 'react-native';
import PropTypes from 'prop-types';

import { withTheme } from '../contexts/theme';
import StyledText from './StyledText';

const styles = StyleSheet.create({
  paragraph: {
    fontSize: 15,
    fontWeight: '400',
    lineHeight: 21,
    marginBottom: 20
  }
});

class Paragraph extends Component {
  render() {
    const { style, theme } = this.props;

    return (
      <StyledText style={[styles.paragraph, theme.paragraph, style]}>
        {this.props.children}
      </StyledText>
    );
  }
}

Paragraph.propTypes = {
  style: PropTypes.any,
  children: PropTypes.node,
  theme: PropTypes.object
};

export default withTheme(Paragraph);
