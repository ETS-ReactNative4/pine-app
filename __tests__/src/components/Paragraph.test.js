import React from 'react';
import { Text } from 'react-native';
import renderer from 'react-test-renderer';
import Paragraph from '../../../src/components/Paragraph';

describe('Paragraph', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <Paragraph>
        <Text>111b4b5f-5b3a-46ef-9ffb-7e6c67e56f78</Text>
      </Paragraph>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
