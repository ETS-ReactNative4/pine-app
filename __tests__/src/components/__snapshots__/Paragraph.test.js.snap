// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paragraph renders correctly 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#000000",
        "fontFamily": "System",
        "fontWeight": "400",
      },
      Array [
        Object {
          "color": "#AAAAAA",
          "fontSize": 15,
          "fontWeight": "400",
          "letterSpacing": -0.24,
          "lineHeight": 21,
          "marginBottom": 20,
        },
        undefined,
      ],
    ]
  }
>
  <Text>
    111b4b5f-5b3a-46ef-9ffb-7e6c67e56f78
  </Text>
</Text>
`;
